README.md

```diff
@@ -1 +1,82 @@
-Paketo Composer Buildpack
+# PHP Composer Distribution Cloud Native Buildpack
+
+This buildpack provides the [composer](https://getcomposer.org/) dependency by installing the composer binary
+onto the build image `$PATH` which makes it available to subsequent buildpacks.
+This buildpack will not issue commands to `composer` (such as `composer install`).
+
+A usage example can be found in the
+[`samples` repository under the `php/composer` directory](https://github.com/paketo-buildpacks/samples/tree/main/php/composer).
+
+## Detection
+
+Will add these requires/provisions to the build plan if and only if a `composer.json` file is found.
```

Comment:
I think the typical detection pattern we use is the `dist` buildpack *always* detects usually doesn't check for the presence of a file (like `composer.json`), because the presence of that file is not needed to install the tool. Then the `install` buildpack that comes next usually detects depending on the availability of the file (`composer.json`), since that buildpack will be the one actually using the file during build (to run `composer install ...`). Since the `install` buildpack `requires` the `dist` buildpack, the `dist` buildpack will only get pulled in when `install`'s  detection criteria is met.

Reply:
Examples of this would be in the Paketo Bundler/Bundle Install buildpacks, and in the Paketo Yarn/Yarn Install buildpacks

Reply:
Does that mean this buildpack always has to be marked `optional`? I'll do this to match the other language families but it seems counterintuitive.

Reply:
I see a few different patterns here. The `poetry` buildpack does check for the existence of `pyproject.toml` or it will return an error about file not existing.

Reply:
@thitch97 can you speak to the poetry case?

Reply:
Josh can you elaborate about where you're seeing these buildpacks marked as optional in this pattern? Subsequent buildapcks (install buildpack) would be responsible for marking this buildpack as optional in it's requirements. I can't see a scenario where the `composer-install` buildpack would optionally require `composer`. It would always need it.

Reply:
> The `poetry` buildpack does check for the existence of `pyproject.toml` or it will return an error about file not existing. https://github.com/paketo-buildpacks/poetry/blob/6f6944571655527a2811101ab19011ae1fd4ad57/detect.go#L24


To me, the `poetry` case is a bit different. [It didn't originally check for the presence of `pyproject.toml`](https://github.com/paketo-buildpacks/poetry/blob/79c651489f24d507a3124e8923eebef5f7480f8a/detect.go) and, though [it does now](https://github.com/paketo-buildpacks/poetry/blob/fe4f3928cec72b723d848004c3dd627580664f83/detect.go#L24), it is implicit and (as I undertand it) in service of honouring the user-set `cpython` version. It was decided in the `poetry` case that the buildpack's operation should not proceed with a mismatched `python` version, making `pyproject.toml` crucial for the functioning of the buildpack itself. In this case, `composer.json` provides no mandatory constraints that would be relevant for the selection of a `composer` version or its subsequent installation.

> `poetry` will always provide `poetry` (if it detects `pyproject.toml`) but will additionally require `poetry` when `BP_POETRY_VERSION` is set. Does that mean the buildpack would be completely independent of other buildpacks when `BP_POETRY_VERSION` is set?

The `poetry` buildpack would still require `cpython` and `pip` and so would not be completely independent. There are other buildpacks, such as `node-engine`, that can be independent.

Reply:
Got it - this makes more sense in the context of the other discussion about `BP_COMPOSER_VERSION`. I'll remove the conditional detect.